src/test/components/Viewport.test.js

```diff
@@ -173,7 +181,33 @@ describe('Viewport', function() {
         // |-----------*------------|
         //             ^
         //             Mouse position
-        it('zooms the preview selection equally when the mouse is centered', () => {
+        it('ctrl zooms the preview selection equally when the mouse is centered', () => {
```

Comment:
So it looks like that test function is only being used on the first test, and the rest of the tests remain duplicated. I think it would be more maintainable here to continue this pattern throughout. If it's too verbose with naming a function every time, you could inline this tester function into a single test, or do a `for of` loop. Here is an example:

```diff
diff --git a/src/test/components/Viewport.test.js b/src/test/components/Viewport.test.js
index 0dd46aec..9f27d73d 100644
--- a/src/test/components/Viewport.test.js
+++ b/src/test/components/Viewport.test.js
@@ -151,6 +151,7 @@ describe('Viewport', function() {
   });
 
   describe('mousewheel zooming using shift + mousewheel', function() {
+    const modifierKeys = [{ ctrlKey: true }, { shiftKey: true }];
     // These series of tests are very particular to margin differences, so check
     // different margin combinations separately.
     const marginTests = [
@@ -174,29 +175,34 @@ describe('Viewport', function() {
         //             ^
         //             Mouse position
         it('zooms the preview selection equally when the mouse is centered', () => {
-          const { scrollAndGetViewport } = setup({
-            marginLeft,
-            marginRight,
-          });
-
-          // Note that clientX will get rounded, so it won't be _exactly_ at the
-          // center, which is why we do some approximations later.
-          const viewport = scrollAndGetViewport({
-            deltaY: -100,
-            shiftKey: true,
-            clientX: BOUNDING_BOX_LEFT + innerComponentWidth * 0.5 + marginLeft,
-          });
-
-          // Assert that this zooms in equally.
-          expect(viewport.viewportLeft).toBeGreaterThan(0);
-          expect(viewport.viewportRight).toBeLessThan(1);
-          expect(viewport.viewportLeft + viewport.viewportRight).toBeCloseTo(1);
-
-          // Only do an additional viewport top/bottom check here.
-          expect(viewport).toMatchObject({
-            viewportTop: 0,
-            viewportBottom: BOUNDING_BOX_HEIGHT,
-          });
+          for (const modifierKey of modifierKeys) {
+            const { scrollAndGetViewport } = setup({
+              marginLeft,
+              marginRight,
+            });
+
+            // Note that clientX will get rounded, so it won't be _exactly_ at the
+            // center, which is why we do some approximations later.
+            const viewport = scrollAndGetViewport({
+              deltaY: -100,
+              clientX:
+                BOUNDING_BOX_LEFT + innerComponentWidth * 0.5 + marginLeft,
+              ...modifierKey,
+            });
+
+            // Assert that this zooms in equally.
+            expect(viewport.viewportLeft).toBeGreaterThan(0);
+            expect(viewport.viewportRight).toBeLessThan(1);
+            expect(viewport.viewportLeft + viewport.viewportRight).toBeCloseTo(
+              1
+            );
+
+            // Only do an additional viewport top/bottom check here.
+            expect(viewport).toMatchObject({
+              viewportTop: 0,
+              viewportBottom: BOUNDING_BOX_HEIGHT,
+            });
+          }
         });
 
         // |        Viewport        |
```